File: app/src/main/java/com/example/lostandfound/ui/ForgotPassword/ForgotPasswordViewModel.java
package com.example.lostandfound.ui.ForgotPassword;

import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

public class ForgotPasswordViewModel extends ViewModel {

    // store the textview used to display error
    private final MutableLiveData<String> forgotPasswordError = new MutableLiveData<>();

    public MutableLiveData<String> getForgotPasswordError(){
        return this.forgotPasswordError;
    }

    public void setForgotPasswordError(String s){
        this.forgotPasswordError.setValue(s);
    }


    // method to validate if email provided is empty
    public boolean validateEmail(String email){
        if (email.isEmpty()){
            forgotPasswordError.setValue("Email cannot be empty");
            return false;
        }
        return true;
    }
}

File: app/src/main/java/com/example/lostandfound/ui/ForgotPassword/ForgotPasswordViewModel.kt
package com.example.lostandfound.ui.ForgotPassword

import androidx.compose.runtime.MutableState
import androidx.compose.runtime.mutableStateOf
import androidx.lifecycle.ViewModel

class ForgotPasswordViewModel : ViewModel() {
    val email: MutableState<String> = mutableStateOf("")
    val error: MutableState<String> = mutableStateOf("")

    fun onEmailChanged(s: String) {
        email.value = s
    }

    fun validateEmail(): Boolean {
        error.value = ""

        if (email.value.trim().isEmpty()) {
            error.value = "Email cannot be empty"
            return false

        } else if (!email.value.contains("@") || !email.value.endsWith("warwick.ac.uk")) {
            error.value = "DevData must be university email (@warwick.ac.uk)"
            return false
        }

        return true
    }
}